File: RightAngle.php
<?php


class ThumbnailTileFullScreenGenerator {


	const	DESIRED_HEIGHT = 5;

	const CONTENT_WIDTH = 10;	

	public $object;

	function __construct($object) {
		$this->object = $object;
	}
  

  function calculateTopAngle() {
  	$bottomAngle = atan2($this->object['height'], $this->object['width'] ) * ( 180 / pi() );
   	return round(90 - $bottomAngle,2);
  }

	function caculateImaginaryImgWidthUsingDesiredHeight() {
		return round(tan(($this->calculateTopAngle() * 2 * 3.14)/360) * self::DESIRED_HEIGHT, 2);
	}

	function calculateHeightUsingContentWidth($actual_width) {
		return round(self::DESIRED_HEIGHT * self::CONTENT_WIDTH/$actual_width,2);
	}


  function findBlockHeight($w) {
  	return round(tan(($this->calculateBlockAngle($w) * 2 * 3.14)/360) * self::CONTENT_WIDTH, 2);
  }

  private function calculateBlockAngle($w) {
  	$bottomAngle = atan2(self::DESIRED_HEIGHT, $w) * ( 180 / pi() );
   	return round($bottomAngle,2);
  }



}

$object = array('height' => 3, "width" => 7);
$calc = new ThumbnailTileFullScreenGenerator($object);
$i = $calc->caculateImaginaryImgWidthUsingDesiredHeight();



$object = array('height' => 6, "width" => 4);
$calc = new ThumbnailTileFullScreenGenerator($object);
$i += $calc->caculateImaginaryImgWidthUsingDesiredHeight();



$object = array('height' => 2, "width" => 3);
$calc = new ThumbnailTileFullScreenGenerator($object);
$i += $calc->caculateImaginaryImgWidthUsingDesiredHeight();




$object = array('height' => 7, "width" => 6);
$calc = new ThumbnailTileFullScreenGenerator($object);
$i += $calc->caculateImaginaryImgWidthUsingDesiredHeight();

echo $calc->findBlockHeight($i);




class Shape {



}



class RightTriangle  extends Shape {
	
	const	DESIRED_HEIGHT = 5;

	const CONTENT_WIDTH = 10;	

	public $object;

	function __construct($object) {
		$this->object = $object;
	}

	function caculateImaginaryImgWidthUsingDesiredHeight() {
		return round(tan(($this->calculateTopAngle() * 2 * 3.14)/360) * self::DESIRED_HEIGHT, 2);
	}

  function calculateHeightUsingContentWidth($actual_width) {
		return round(self::DESIRED_HEIGHT * self::CONTENT_WIDTH/$actual_width,2);
	}

  function findBlockHeight($w) {
  	return round(tan(($this->calculateBlockAngle($w) * 2 * 3.14)/360) * self::CONTENT_WIDTH, 2);
  }

  private function calculateTopAngle() {
  	$bottomAngle = atan2($this->object['height'], $this->object['width'] ) * ( 180 / pi() );
   	return round(90 - $bottomAngle,2);
  }

  private function calculateBlockAngle($w) {
  	$bottomAngle = atan2(self::DESIRED_HEIGHT, $w) * ( 180 / pi());
   	return round($bottomAngle,2);
  }

}



?>

File: RightAngle.rb
#!/usr/bin/ruby


#Small script used to calculate thumb height so that all thumbs on 
#same line hold aspect ratio but fill up hole area, like google have on image search.
# I got various height of thumbs so first give them desired height,
# then loop  over the thumb doing addition of calculated width untill reach approximately allowed width 
# then I have same height and approx allowed width.
# then calculate angle for hole block 
# then calculate height for allowed width with given angle
# when got it just set it up for thumbs in line, width set to auto, and that is it. 
class Shape



end


class RightTriangle < Shape

	DESIRED_HEIGHT = 30

	ALLOWED_WIDTH = 800

	attr_accessor :height, :width

	def initialize(hash) 
		@height = hash[:height].to_f
		@width = hash[:width].to_f
	end	
    
    #In a right triangle, 
    #the tangent of an angle 
    #is the length of the opposite 
    #side divided by the length of the adjacent side
	def top_angle
		return Math.atan2(@width,@height) * ( 180 / Math::PI )
	end	

	def bottom_angle
		return 90 - top_angle
	end	

	def calculate_width_using_desired_height
		return Math.tan((self.top_angle * 2 * 3.14)/360) * RightTriangle::DESIRED_HEIGHT
	end

	def calculate_shared_top_angle(elements_width)
		return Math.atan2(elements_width, RightTriangle::DESIRED_HEIGHT) * ( 180 / Math::PI )
	end	

	def calculate_shared_bottom_angle(elements_width)
		return 90 - Math.atan2(elements_width, RightTriangle::DESIRED_HEIGHT) * ( 180 / Math::PI )
	end	

	def calculate_shared_height(elements_width)
  		return Math.tan((self.calculate_shared_bottom_angle(elements_width) * 2 * 3.14)/360) * RightTriangle::ALLOWED_WIDTH;
	end	

end


		

@hash = {:height => 40, :width => 50 }

@triangle = RightTriangle.new(@hash)


#puts @triangle.instance_variables

puts @triangle.top_angle
puts @triangle.bottom_angle

puts @triangle.calculate_width_using_desired_height

puts @triangle.calculate_shared_top_angle(400)

puts @triangle.calculate_shared_bottom_angle(400)

puts @triangle.calculate_shared_height(400)



	